src/hdmf/common/resources.py

```diff
@@ -165,6 +166,59 @@ def __init__(self, **kwargs):
         self.object_keys = kwargs['object_keys'] or ObjectKeyTable()
         self.type_map = kwargs['type_map'] or get_type_map()
 
+    @staticmethod
+    def assert_external_resources_equal(left, right, check_dtype=True):
+        """
+        Compare that the keys, resources, entities, objects, and object_keys tables match
+
+        :param left: ExternalResources object to compare with right
+        :param right: ExternalResources object to compare with left
+        :param check_dtype: Enforce strict checking of dtypes. Dtypes may be different
+            for example for ids, where depending on how the data was saved
+            ids may change from int64 to int32. (Default: True)
+        :returns: The function returns True if all values match. If mismatches are found,
+            AssertionError will be raised.
+        :raises AssertionError: Raised if any differences are found. The function collects
+            all differences into a single error so that the assertion will indicate
+            all found differences.
+        """
+        errors = []
+        try:
+            pd.testing.assert_frame_equal(left.keys.to_dataframe(),
+                                          right.keys.to_dataframe(),
+                                          check_dtype=check_dtype)
+        except AssertionError as e:
```

Comment:
It may seem silly, but can you test each of these error cases? In general, there may be an error in the exception handling (probably not here though). We should strive for 100% code coverage on new changes to HDMF, given that our goal is to get to 100% code coverage for the whole repo.

Reply:
Sure